taichi/transforms/type_check.cpp

```diff
@@ -94,7 +96,7 @@ class TypeCheck : public IRVisitor {
       TI_WARN(
           "[{}] Local store may lose precision (target = {}, value = {}, at",
           stmt->name(), stmt->ptr->ret_data_type_name(),
-          stmt->data->ret_data_type_name(), stmt->id);
+          old_data->ret_data_type_name(), stmt->id);
```

Comment:
The error message was like `can't store i32 to i32`, since `stmt->data` is already updated to the type-cast version...

Reply:
Thank for the OFT, it confused me a long time..